### PR TITLE
Railgun

### PR DESCRIPTION
**Pull request by BaegEhl**
adds a railgun weapon
note: this obviously wasn't requested, I just felt like doing it and figured I might as well push it in case it actually meets the standards of the game

### CODE REVIEW

File: Assets/Scripts/Functional Definitions/Abilities/Railgun.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                line.startWidth -= 0.01F;
+                line.endWidth -= 0.01F;
+            }
+            if(line.startWidth < 0)
+            {
+                line.startWidth = line.endWidth = 0;
+                firing = false;
+            }
+
+        } else if (timer != -9999f) //this is a really stupid bandaid but it works so i'm calling it a feature
```
Comment (rudderbucky) [inline Assets/Scripts/Functional Definitions/Abilities/Railgun.cs:70]: Preferably don't do this

**Comment by rudderbucky** [comment]
I don't think I like the idea of a delayed beam. The Ion shooter looks nice though, we could just use that for Ion

**Comment by BaegEhl** [comment]
> I don't think I like the idea of a delayed beam. The Ion shooter looks nice though, we could just use that for Ion

this was originally intended to be an ion beam sprite entry, I reused it for this since it made sense. I think the ion version should be a bit bigger though since it's a larger beam

**Comment by rudderbucky** [comment]
I don't think there is a pressing need for more weapon abilities at this point.

**Pull request closed without merging**